agent/service.go

```diff
@@ -79,6 +112,73 @@ func (ts *TaskService) Create(ctx context.Context, req *shimapi.CreateTaskReques
 	return resp, nil
 }
 
+func (ts *TaskService) proxyStdio(ctx context.Context, stdin, stdout, stderr io.ReadWriteCloser) error {
+	// Setup the vsock connection, once received start reading and writing to stream
+	out := func(reader io.ReadWriteCloser, port uint32) {
+		if reader == nil {
+			return
+		}
+		listener, err := vsock.Listen(port)
+		log.G(ctx).Debug("Created listener")
+		if err != nil {
+			log.G(ctx).WithError(err).Error("unable to listen on vsock")
+			return
+		}
+		conn, err := listener.Accept()
+		log.G(ctx).Debug("Accepted connection")
+		if err != nil {
+			log.G(ctx).WithError(err).Error("unable to accept vsock connection")
+			return
+		}
+
+		buf := make([]byte, internal.DefaultBufferSize)
+		for {
+			read, err := reader.Read(buf)
+			_, werr := conn.Write(buf[:read])
```

Comment:
It looks like you're ignoring both `err` and `werr`, so the loop here is infinite.  Do we want the loop to be infinite or is there a reasonable exit condition?

Reply:
Right now the exit conditions is "the container exits, causing the process to exit". There could probably be a better way to signal that we are good to exit though and wind this down gracefully.

It might make sense to wait on these for a tiny bit of time before calling os.Exit() in shutdown?